test/integration/testPutGet.js

```diff
@@ -857,3 +861,315 @@ describe('PUT GET test multiple files', function () {
+                    });
+                    stream.on('data', function (row) {
+                      assert.strictEqual(row['status'], DOWNLOADED);
+                      // Verify the size of the file
+                      const file = path.join(tmpDir, row.file);
+                      const size = fs.statSync(file).size;
+                      assert.strictEqual(size, testCase.size);
+                    });
+                    stream.on('end', function () {
+                      callback();
+                    });
+                  }
+                }
+              });
+            }
+          ],
+          done
+        );
+      }
+    };
+  };
+
+  for (let index = 0; index < testCases.length; index++) {
+    const testCase = testCases[index];
+    it(testCase.name, createItCallback(testCase));
+  }
+});
+
+describe('PUT GET test with error', function () {
+  let connection;
+  const stage = `@${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+  const stage_not_exist = `@${DATABASE_NAME}.${SCHEMA_NAME}.%NONEXISTTABLE`;
+  const createTable = `create or replace table ${TEMP_TABLE_NAME} (${COL1} STRING, ${COL2} STRING, ${COL3} STRING)`;
+  const removeFile = `REMOVE ${stage}`;
+  const dropTable = `DROP TABLE IF EXISTS ${TEMP_TABLE_NAME}`;
+
+  // Create a temp file without specified file extension
+  const tmpFile = tmp.fileSync(); 
+
+  before(async () => {
+    connection = testUtil.createConnection();
+    await testUtil.connectAsync(connection);
+    await testUtil.executeCmdAsync(connection, createTable);
+  });
+
+  after(async () => {
+    testUtil.deleteFileSyncIgnoringErrors(tmpFile);
+
+    await testUtil.executeCmdAsync(connection, removeFile);
+    await testUtil.executeCmdAsync(connection, dropTable);
+    await testUtil.destroyConnectionAsync(connection);
+  });
+
+  const tmpfilePath = getPlatformTmpPath(tmpFile.name);
+
+  const testCases =
+    [
+      {
+        name: 'file not exist',
+        putQuery: `PUT file_not_exist.txt ${stage}  OVERWRITE=true`,
+      },
+      {
+        name: 'compresssed file not exist',
+        putQuery: `PUT file_not_exist.gzip ${stage}  OVERWRITE=true`,
+      },
+      {
+        name: 'stage not exist',
+        putQuery: `PUT ${tmpfilePath} ${stage_not_exist}`,
+      },
+    ];
+
+  const createItCallback = function (testCase) {
+    return function (done) {
+      {
+        async.series(
+          [
+            function (callback) {
+              connection.execute({
+                sqlText: testCase.putQuery,
+                complete: function (err) {
+                  if(err){
+                    assert.strictEqual(err.data.type, 'COMPILATION');
+                    callback();
+                  }else {
+                    callback(err);
+                  }
+                }
+              });
+            }
+          ],
+          done
+        );
+      }
+    };
+  };
+
+  for (let index = 0; index < testCases.length; index++) {
```

Comment:
can we favor foreach here (as I wrote in comment above also)?

Reply:
updated.